mobile_verifier/src/boosting_oracles/mod.rs

```diff
@@ -1,136 +1,418 @@
+        Ok(HexAssignments {
+            footfall,
+            urbanized,
+            landtype,
+        })
+    }
 }
 
-pub struct HexBoostData<Urban, Foot> {
-    pub urbanization: Urban,
-    pub footfall: Foot,
-}
+impl HexBoostData {
+    fn urbanized_assignment(&self, cell: hextree::Cell) -> anyhow::Result<Assignment> {
+        if !self.usa_geofence.in_valid_region(&cell) {
+            return Ok(Assignment::C);
+        }
 
-pub struct UrbanizationData<Urban, Geo> {
-    urbanized: Urban,
-    usa_geofence: Geo,
-}
+        match self.urbanized.get(cell)?.is_some() {
+            true => Ok(Assignment::A),
+            false => Ok(Assignment::B),
+        }
+    }
 
-pub struct FootfallData<Foot> {
-    footfall: Foot,
-}
+    fn footfall_assignment(&self, cell: hextree::Cell) -> anyhow::Result<Assignment> {
+        let Some((_, vals)) = self.footfall.get(cell)? else {
+            return Ok(Assignment::C);
+        };
 
-impl<Urban, Foot> HexBoostData<Urban, Foot> {
-    pub fn new(urbanization: Urban, footfall: Foot) -> Self {
-        Self {
-            urbanization,
-            footfall,
+        match vals {
+            &[x] if x >= 1 => Ok(Assignment::A),
+            &[0] => Ok(Assignment::B),
+            other => anyhow::bail!("unexpected footfall disktree data: {cell:?} {other:?}"),
         }
     }
-}
 
-impl<Urban, Geo> UrbanizationData<Urban, Geo> {
-    pub fn new(urbanized: Urban, usa_geofence: Geo) -> Self {
-        Self {
-            urbanized,
-            usa_geofence,
-        }
+    fn landtype_assignment(&self, cell: hextree::Cell) -> anyhow::Result<Assignment> {
+        let Some((_, vals)) = self.landtype.get(cell)? else {
+            return Ok(Assignment::C);
+        };
+
+        anyhow::ensure!(
+            vals.len() == 1,
+            "unexpected landtype disktree data: {cell:?} {vals:?}"
+        );
+
+        let cover = Landtype::try_from(vals[0])?;
+        Ok(cover.into())
     }
 }
 
-impl<Foot> FootfallData<Foot> {
-    pub fn new(footfall: Foot) -> Self {
-        Self { footfall }
+impl From<Landtype> for Assignment {
+    fn from(value: Landtype) -> Self {
+        match value {
+            Landtype::Built => Assignment::A,
+            //
+            Landtype::Tree => Assignment::B,
+            Landtype::Shrub => Assignment::B,
+            Landtype::Grass => Assignment::B,
+            //
+            Landtype::Bare => Assignment::C,
+            Landtype::Crop => Assignment::C,
+            Landtype::Frozen => Assignment::C,
+            Landtype::Water => Assignment::C,
+            Landtype::Wet => Assignment::C,
+            Landtype::Mangrove => Assignment::C,
+            Landtype::Moss => Assignment::C,
+        }
     }
 }
 
-trait DiskTreeLike: Send + Sync {
-    fn get(&self, cell: hextree::Cell) -> hextree::Result<Option<(hextree::Cell, &[u8])>>;
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum Landtype {
```

Comment:
Nit: LandType rather than Landtype

Reply:
I ended up with `Landtype` for a few reasons. 

I my opinion, as a member, `landtype` reads nicer to me than `land_type`. 
Especially in places where you're qualifying the data
```rs
let landtype_disktree = DiskTreeMap::open(&settings.landtype_data_set)?;
```
And it felt like it fit in more with `footfall` and `urbanized`. 

When I say them out loud with the inflection of capitalization, both `Landtype` and `LandType` feel more or less the same. So I went with what I thought led to more aesthetically pleasing code.